src/WebJobs.Extensions.DurableTask/IHostLifetime.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Threading;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Service interface for signaling the extension when the host is starting up or shutting down.
+    /// </summary>
+    /// <remarks>
+    /// This interface is expected to be used as an injected service.
+    /// </remarks>
+    public interface IHostLifetime
```

Comment:
Hmm...I know there was another interface for this but it had a different name. Any suggestions on how to best avoid confusion? I considered IFunctionHostLifetime but that could still be confused with something that's defined by the Functions runtime.

Reply:
Is there a reason we even have this custom interface? It seems like our only implementation is just a wrapper for `IApplicationLifetime`, and I don't immediately see the gain in that wrapper.

Reply:
It was to avoid the complexity of introducing an interface that is available only in `netstandard2.0`. Having my own interface essentially reduced the number of scope of `#if` statements I needed throughout the code (I think test code was also implicated in this).

Reply:
I see... that makes sense. Can you add this detail to the summary section? Maybe consider a name like `ICrossFunctionsVersionAppLifeTime` to indicate why we have a new interface?

In general, this is less of an issue in Visual Studio, because I could have dove into the interface and saw that it was a custom interface, not the .NET Standard interface.

Reply:
I'll rename it to `IApplicationLifetimeWrapper`. I don't want to make it more descriptive than that because it could mislead people into thinking it has more complex semantics than the underlying `IApplicationLifetime`.